functions.php

```diff
@@ -0,0 +1,48 @@
+<?php
+    session_start();
+    //render the csrf token input field
+    function csrfToken()
+    {
+        if(function_exists("random_bytes"))
+        {
+            $token = $_SESSION['token'] = bin2hex(random_bytes(20));
```

Comment:
[`random_bytes()`](https://www.php.net/manual/en/function.random-bytes.php) may also return exceptions which are not handled and can have predictable output if / when they occur.

Reply:
I'm guessing you're referring to the wiki's following quote: <br>`If an appropriate source of randomness cannot be found, an Exception will be thrown.`
Correct?

Reply:
Since the number `20` is valid and tested, this seems to be the only case. I know it's unlikely to happen and adding a few more lines of code to handle that can be frustrating, especially given all the rest of the code, but it's something I found. I don't think it definitely has to be fixed, it's a good to have ;)

Reply:
Sure thing, I'll add it nonetheless.